### ISSUE TITLE
read() throws RuntimeError

### ISSUE THREAD

**Issue by jcd496** [ornladios/ADIOS2#2356]
**Describe the bug**
read() from large directory throws RuntimeError, behaves correctly on smaller directory.
```
RuntimeError: ERROR: couldn't open file /path/to/checkpoint2.bp.dir//checkpoint2.bp.1019,, check permissions or path existence, in call to POSIX open
: iostream error
```

I have permission to access the checkpoint2.bp and each dir in the path.  
When I run on checkpoint1.bp, with checkpoint1.bp.dir/ being smaller than checkpoint2.bp.dir/, read() behaves correctly.

This issue seems related to open issue:   Reading a file with many subfiles fails #1906

**To Reproduce**
```
import numpy as np
import adios2

with adios2.open('/path/to/checkpoint2.bp', 'r') as fh:
    x = fh.read("mprts::mprts::size_by_patch")
```

**Expected behavior**
I would expect a numpy array to be stored in x
I tried using both fh.read, and stepping through the file with `for fstep in fh:`, as demonstrated in the docs.

**Desktop (please complete the following information):**
 - OS/Platform: Summit
 - Build: conda install -c conda-forge adios2   
 - Veresion: 2.5.0

**Additional context**
I noticed that after performing conda install -c conda-forge adios2, there is a clash which causes mpi4py to no longer work on the conda env and mpi runs to fail.   This occurs when using jsrun, before installing adios2 mpi4py behaves as expected.

Thanks!

**Following up**
Was the issue fixed? Please report back.

**Comment by pnorbert** [comment]
Hi,

Can you raise the limit of the number of opened files with `ulimit -n 2048` or whatever the number of files in the directory is?
Alternatively, could you read this data in parallel, so that no single process needs to open a thousand subfiles?
If the data is important and none of the above works, you may try to use adios2_reorganize on M processes to read back this data and write it out in M files (only works for global arrays, not for local arrays).

For the conda mpi4py issue, can you try using adios 2.6.0 and report in a new issue if it still gives you trouble?

**Comment by jcd496** [comment]
Hi,
The data is large but by running an interactive job on Summit I was able to set ulimit -n higher than authorized on the login nodes and then ran adios_reorganize  to redistribute the data across fewer subfiles.  
thanks!
